### PR TITLE
Editor: try compact notice for categories

### PR DESCRIPTION
**Pull request by timmyc**
This is a follow-up item from #7247 - specifically, in that branch a notice was added that is shown when the Jetpack version does not satisfy the minimum requirement for the taxonomy endpoints.  This branch changes up the current look with the `isCompact` on the `<Notice />`

__Before__
new_post_ _major_keys_and_monads_ _wordpress_com

__After__
new_post_ _major_keys_and_monads_ _wordpress_com

__To Test__
In order to test this out, you must have a Jetpack site running a version older than `4.1.0`, and then open a post in the editor for that site.

Test live: https://calypso.live/?branch=update/editor/jp-update-notice

### CODE REVIEW

**Comment by folletto** [comment]
What's the reason for changing to the small version? The bigger one seems a little more appropriate here, and seems behaving better with the line breaks.

**Comment by timmyc** [comment]
@folletto I suppose this was a bit of a "try" branch based upon some feedback @aduth left during the review of #7247

**Comment by aduth** [comment]
What I had in mind was more like what we do for notices with actions in smaller viewports:

Notice

(From [`/devdocs/design/notices`](https://wpcalypso.wordpress.com/devdocs/design/notices))

**Comment by folletto** [comment]
> What I had in mind was more like what we do for notices with actions in smaller viewports:

That is more fitting 👍

**Comment by timmyc** [comment]
Thanks for jogging the memory there @aduth.  In 46fe3a0 I added a new prop to `<Notice />` called `isSmallViewport`.  That then applies the `.is-small-viewport` classname which "forces" the <480px logic.

I'm unsure of if the className and prop is the most logical choice, but the end result is:

new_post_ _major_keys_and_monads_ _wordpress_com

**Comment by aduth** [comment]
The class name does feel a bit forced, especially since this context clearly highlights how it can be a useful layout outside of just small viewport devices. I wonder if we might be jumping the gun on abstracting this as a feature of `<Notice />` (despite having been the one to suggest it) and maybe we ought to just apply the styles with a selector specific to the editor sidebar.

Else, for naming might consider: "stacked", "vertical", or something to do with small width.

I wonder if it'd be possible to detect this from the component itself, i.e. measure the width when the component mounts (opens questions on what to do when window is resized though, which gets ugly quick).

File: client/components/notice/index.jsx [hunk 1]
```diff
@@ -108,6 +109,10 @@ export default React.createClass( {
 			noticeClass = classnames( noticeClass, 'is-compact' );
 		}
 
+		if ( this.props.isSmallViewport ) {
+			noticeClass = classnames( noticeClass, 'is-small-viewport' );
```
Comment (aduth) [inline client/components/notice/index.jsx]: This is more a general refactoring point for the whole component, but it's more concise, clean, and performant to pass optional modifiers as the second object argument:

```js
const noticeClass = classnames( 'notice', this.props.status, this.props.className, {
	'is-compact': this.props.isCompact,
	'is-small-viewport': this.props.isSmallViewport,
	'is-dismissable': this.props.showDismiss
} );
```

**Comment by timmyc** [comment]
Removed the updates to `<Notice />` and just applied the styling within the component here instead.

**Comment by aduth** [comment]
Tested by downgrading my Jetpack version to 4.0.4 and verified notice appears as expected, replaced with categories selection after upgrading. 👍 

Assuming https://github.com/Automattic/wp-calypso/pull/7532#issuecomment-242779633 as design sign-off, marking this ready to merge.

**Pull request merged**